[Bug]: Optimism swaps - Error fetching quotes

### Describe the bug

I cannot fetch quotes on Optimism, Ethereum Mainnet and Arbitrum still work

https://user-images.githubusercontent.com/53189696/233144278-1c8b5e7f-22da-4eeb-a343-17feeb95ffbb.mov



### Steps to reproduce

1. Log in to the extension
2. Switch to optimism
3. click swaps on the overview page
4. enter an eth amount
5. change the to currency to usdc
6. click review swap


### Error messages or log output

```shell
swaps.js:837 Error fetching quotes:  Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="getL1Fee(bytes)", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0)
  at e.makeError (index.js:238:1)
  at e.throwError (index.js:247:1)
  at e.decodeFunctionResult (interface.js:388:1)
  at t.<anonymous> (index.js:395:1)
  at index.js:48:1
  at Object.next (index.js:29:1)
  at a (index.js:20:1)
```


### Version

10.29

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

--- a/app/scripts/controllers/swaps.js
+++ b/app/scripts/controllers/swaps.js
@@ -113,7 +113,7 @@ export default class SwapsController {
     fetchTradesInfo = defaultFetchTradesInfo,
     getCurrentChainId,
     getEIP1559GasFeeEstimates,
-    onNetworkDidChange,
+    onNetworkStateChange,
   }) {
     this.store = new ObservableStore({
       swapsState: { ...initialState.swapsState },
@@ -137,7 +137,7 @@ export default class SwapsController {
 
     this.ethersProvider = new Web3Provider(provider);
     this._currentNetworkId = networkController.store.getState().networkId;
-    onNetworkDidChange(() => {
+    onNetworkStateChange(() => {
       const { networkId, networkStatus } = networkController.store.getState();
       if (
         networkStatus === NetworkStatus.Available &&

--- a/app/scripts/controllers/swaps.test.js
+++ b/app/scripts/controllers/swaps.test.js
@@ -160,7 +160,7 @@ describe('SwapsController', function () {
     return new SwapsController({
       getBufferedGasLimit: MOCK_GET_BUFFERED_GAS_LIMIT,
       networkController: getMockNetworkController(),
-      onNetworkDidChange: sinon.stub(),
+      onNetworkStateChange: sinon.stub(),
       provider: _provider,
       getProviderConfig: MOCK_GET_PROVIDER_CONFIG,
       getTokenRatesState: MOCK_TOKEN_RATES_STORE,
@@ -208,11 +208,11 @@ describe('SwapsController', function () {
 
     it('should replace ethers instance when network changes', function () {
       const networkController = getMockNetworkController();
-      const onNetworkDidChange = sinon.stub();
+      const onNetworkStateChange = sinon.stub();
       const swapsController = new SwapsController({
         getBufferedGasLimit: MOCK_GET_BUFFERED_GAS_LIMIT,
         networkController,
-        onNetworkDidChange,
+        onNetworkStateChange,
         provider,
         getProviderConfig: MOCK_GET_PROVIDER_CONFIG,
         getTokenRatesState: MOCK_TOKEN_RATES_STORE,
@@ -220,7 +220,7 @@ describe('SwapsController', function () {
         getCurrentChainId: getCurrentChainIdStub,
       });
       const currentEthersInstance = swapsController.ethersProvider;
-      const changeNetwork = onNetworkDidChange.getCall(0).args[0];
+      const changeNetwork = onNetworkStateChange.getCall(0).args[0];
 
       networkController.store.getState.returns({
         networkId: NETWORK_IDS.MAINNET,
@@ -238,11 +238,11 @@ describe('SwapsController', function () {
 
     it('should not replace ethers instance when network changes to loading', function () {
       const networkController = getMockNetworkController();
-      const onNetworkDidChange = sinon.stub();
+      const onNetworkStateChange = sinon.stub();
       const swapsController = new SwapsController({
         getBufferedGasLimit: MOCK_GET_BUFFERED_GAS_LIMIT,
         networkController,
-        onNetworkDidChange,
+        onNetworkStateChange,
         provider,
         getProviderConfig: MOCK_GET_PROVIDER_CONFIG,
         getTokenRatesState: MOCK_TOKEN_RATES_STORE,
@@ -250,7 +250,7 @@ describe('SwapsController', function () {
         getCurrentChainId: getCurrentChainIdStub,
       });
       const currentEthersInstance = swapsController.ethersProvider;
-      const changeNetwork = onNetworkDidChange.getCall(0).args[0];
+      const changeNetwork = onNetworkStateChange.getCall(0).args[0];
 
       networkController.store.getState.returns({
         networkId: null,
@@ -268,11 +268,11 @@ describe('SwapsController', function () {
 
     it('should not replace ethers instance when network changes to the same network', function () {
       const networkController = getMockNetworkController();
-      const onNetworkDidChange = sinon.stub();
+      const onNetworkStateChange = sinon.stub();
       const swapsController = new SwapsController({
         getBufferedGasLimit: MOCK_GET_BUFFERED_GAS_LIMIT,
         networkController,
-        onNetworkDidChange,
+        onNetworkStateChange,
         provider,
         getProviderConfig: MOCK_GET_PROVIDER_CONFIG,
         getTokenRatesState: MOCK_TOKEN_RATES_STORE,
@@ -280,7 +280,7 @@ describe('SwapsController', function () {
         getCurrentChainId: getCurrentChainIdStub,
       });
       const currentEthersInstance = swapsController.ethersProvider;
-      const changeNetwork = onNetworkDidChange.getCall(0).args[0];
+      const changeNetwork = onNetworkStateChange.getCall(0).args[0];
 
       networkController.store.getState.returns({
         networkId: NETWORK_IDS.GOERLI,

--- a/app/scripts/metamask-controller.js
+++ b/app/scripts/metamask-controller.js
@@ -1195,10 +1195,8 @@ export default class MetamaskController extends EventEmitter {
         this.txController.txGasUtil,
       ),
       networkController: this.networkController,
-      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
-        networkControllerMessenger,
-        NetworkControllerEventType.NetworkDidChange,
-      ),
+      onNetworkStateChange: (listener) =>
+        this.networkController.networkIdStore.subscribe(listener),
       provider: this.provider,
       getProviderConfig: () => this.networkController.store.getState().provider,
       getTokenRatesState: () => this.tokenRatesController.state,

--- a/app/scripts/metamask-controller.js
+++ b/app/scripts/metamask-controller.js
@@ -1196,7 +1196,7 @@ export default class MetamaskController extends EventEmitter {
       ),
       networkController: this.networkController,
       onNetworkStateChange: (listener) =>
-        this.networkController.networkIdStore.subscribe(listener),
+        this.networkController.store.subscribe(listener),
       provider: this.provider,
       getProviderConfig: () => this.networkController.store.getState().provider,
       getTokenRatesState: () => this.tokenRatesController.state,

